link/xdp.go

```diff
@@ -0,0 +1,54 @@
+package link
+
+import (
+	"fmt"
+
+	"github.com/cilium/ebpf"
+)
+
+// XDPAttachFlags represents how XDP program will be attached to interface.
+type XDPAttachFlags uint32
+
+const (
+	// XDPGenericMode (SKB) links XDP BPF program for drivers which do
+	// not yet support native XDP.
+	XDPGenericMode XDPAttachFlags = 1 << (iota + 1)
+	// XDPDriverMode links XDP BPF program into the driver’s receive path.
+	XDPDriverMode
+	// XDPOffloadMode offloads the entire XDP BPF program into hardware.
+	XDPOffloadMode
+)
+
+type XDPOptions struct {
+	// Program must be an XDP BPF program.
+	Program *ebpf.Program
+
+	// Interface is the interface index to attach program to.
+	Interface int
+
+	// Flags must match the attach flag of XDP.
```

Comment:
Is that correct? I think you can load XDP without any flags and still force XDPDriverMode when attaching (at least that's possible via netlink).

Reply:
It explained more right after this line:
```
// Only one XDP mode should be set, without flag defaults
// to driver/generic mode (best effort).
```
but I just rephrased it line#29 please let me know if you want it something else.